financial-connections/src/main/java/com/stripe/android/financialconnections/presentation/FinancialConnectionsSheetNativeViewModel.kt

```diff
@@ -449,3 +500,38 @@ internal sealed interface FinancialConnectionsSheetNativeViewEffect {
         val result: FinancialConnectionsSheetActivityResult
     ) : FinancialConnectionsSheetNativeViewEffect
 }
+
+private fun FinancialConnectionsSheetNativeState.toTopAppBarState(
+    forceHideStripeLogo: Boolean,
+): TopAppBarState {
+    return TopAppBarState(
+        hideStripeLogo = reducedBranding,
+        forceHideStripeLogo = forceHideStripeLogo,
+        isTestMode = testMode,
+    )
+}
+
+private fun <K, V> Flow<Map<K, V>>.getValueFromKeyFlow(keyFlow: Flow<K>): Flow<V> {
+    return combineTransform(keyFlow) { map, key ->
+        val result = map[key]
+        if (result != null) {
+            emit(result)
+        }
+    }.distinctUntilChanged()
+}
+
+private fun <K, V> MutableStateFlow<Map<K, V>>.updateWithNewEntry(entry: Pair<K, V>) {
+    update { it + mapOf(entry) }
+}
+
+private fun <K, V> MutableStateFlow<Map<K, V>>.updateWithNewEntry(key: K, transform: (V) -> V) {
+    update {
+        val oldValue = it[key]
+        if (oldValue != null) {
+            val newValue = transform(oldValue)
+            it + mapOf(key to newValue)
+        } else {
+            it
+        }
+    }
+}
```

Comment:
we might want to move these to a flow extensions file.

Reply:
Yep, can do that!